test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -1521,6 +1521,263 @@ public async Task GetTrustResultAsync_WithExpiredRepositoryCertificateAndNoTimes
+                    throw new ArgumentNullException(nameof(fixture));
+                }
+
+                _fixture = fixture;
+                _provider = new SignatureTrustAndValidityVerificationProvider();
+            }
+
+            // Case1: primary signature (trusted + non-expired) doesn't fall back to countersiganture (trusted + non-expired).
+            // The verification result is the primary signature status(valid). 
+            [CIOnlyFact]
+            public async Task GetTrustResultAsync_FallbackFromGoodPrimarySigntuareToGoodCountersignature_NoFallbackAndReturnsValidAsync()
+            {
+                var settings = _defaultNuGetOrgSettings;
+
+                using (Test test = await Test.CreateAuthorSignedRepositoryCountersignedPackageAsync(
+                    authorCertificate: _fixture.TrustedTestCertificate.Source.Cert,
+                    repositoryCertificate: _fixture.TrustedRepositoryCertificate.Source.Cert))
+                using (var packageReader = new PackageArchiveReader(test.PackageFile.FullName))
+                {
+                    PrimarySignature primarySignature = await packageReader.GetPrimarySignatureAsync(CancellationToken.None);
+                    PackageVerificationResult status = await _provider.GetTrustResultAsync(packageReader, primarySignature, settings, CancellationToken.None);
+
+                    Assert.Equal(SignatureVerificationStatus.Valid, status.Trust);
+                }
+            }
+
+            // Case2: primary signature (trusted + non-expired) doesn't fall back to countersiganture untrusted + non-expired).
+            // The verification result is the primary signature status(valid).
+            [CIOnlyFact]
+            public async Task GetTrustResultAsync_FallbackFromGoodPrimarySigntuareToUntrustedCountersignature_NoFallbackAndReturnsValidAsync()
+            {
+                var settings = _defaultNuGetOrgSettings;
+
+                using (Test test = await Test.CreateAuthorSignedRepositoryCountersignedPackageAsync(
+                    authorCertificate: _fixture.TrustedTestCertificate.Source.Cert,
+                    repositoryCertificate: _fixture.UntrustedTestCertificate.Cert))
+                using (var packageReader = new PackageArchiveReader(test.PackageFile.FullName))
+                {
+                    PrimarySignature primarySignature = await packageReader.GetPrimarySignatureAsync(CancellationToken.None);
+                    PackageVerificationResult status = await _provider.GetTrustResultAsync(packageReader, primarySignature, settings, CancellationToken.None);
+
+                    Assert.Equal(SignatureVerificationStatus.Valid, status.Trust);
+                }
+            }
+
+            // Case3: primary signature (untrusted + non-expired) falls back to countersiganture (trusted + non-expired).
+            // The verification result is the sever one of fallback status(valid) and the countersignature status(valid), so it's valid.
```

Comment:
"...the sever one..."?

Can you reword?   Check other test comments for same issue.

Reply:
Sorry! Fixed.